packages/slice-machine/components/FormFields/CheckboxControl.jsx

```diff
@@ -31,7 +32,7 @@ const CheckboxControl = ({
     // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
     buildControlledValue
       ? // eslint-disable-next-line @typescript-eslint/no-unsafe-call
-        setControlledValue(buildControlledValue(controlledValue))
+        setControlledValue(buildControlledValue(controlledValue, isChecked))
       : // eslint-disable-next-line @typescript-eslint/no-unsafe-call
         setControlledValue(buildControlledValue);
```

Comment:
Honestly, I thought the same 😄 
For me, the whole useEffect could be improved and maybe even removed, but since it's jsx with old forms I decided to only fix the bug and assuring myself that I will not break something else.
Do you think I should refactor this now?